### ISSUE TITLE
rifle's root flag does not work with non-POSIX shells

### ISSUE THREAD

**Issue by kmarius** [ranger/ranger#1396]
#### Runtime Environment

ranger version: ranger 1.9.2
Python version: 3.7.1 (default, Oct 22 2018, 10:41:28) [GCC 8.2.1 20180831]
Locale: en_US.UTF-8
SHELL: /usr/bin/fish

#### Current Behavior
The following error appears when opening a file with rifle with the root flag enabled (exit ranger to see it):
```
set: Variable name '/home/kmarius/Documents' is not valid. See `help identifiers`.                                            
Standard input (line 1):
set -- '/home/kmarius/Documents'; echo "$@"
^
       set

   Synopsis
       set [SCOPE_OPTIONS]
       set [OPTIONS] VARIABLE_NAME VALUES...
       set [OPTIONS] VARIABLE_NAME[INDICES]... VALUES...
       set ( -q | --query ) [SCOPE_OPTIONS] VARIABLE_NAMES...
       set ( -e | --erase ) [SCOPE_OPTIONS] VARIABLE_NAME
       set ( -e | --erase ) [SCOPE_OPTIONS] VARIABLE_NAME[INDICES]...                                                      
       set ( -S | --show ) [SCOPE_OPTIONS] [VARIABLE_NAME]...

set: Type “help set” for related documentation

fish: $@ is not supported. In fish, please use $argv.
set -- '/home/kmarius/Documents'; echo "$@"
```
Obviously, the command is wrongly executed with `fish` which doesn't understand `set --` and `$@`  as used in POSIX shells.

#### Possible Solution
Given the `-m` flag (to keep the environment), `su` will run the command with whatever it finds in `$SHELL`, in this case `/usr/bin/fish`. We can tell `su` to use `/bin/sh` by adding  `'-s', '/bin/sh'` to 
https://github.com/ranger/ranger/blob/bae44ae3c7440c00acf65ba4c26a0fd2a0f78c3a/ranger/ext/rifle.py#L358

#### Steps to reproduce
1. add `match ., flag r = echo "$@"` to `rifle.conf` (this will match any file)
2. run ranger with `SHELL=/usr/bin/fish ranger` (or run fish as login shell)
3. open any file with the `:open_with` command (default key `r`) and the entry defined above, quit ranger

**Comment by Vifon** [comment]
I think we should drop `-m` and `-E` and use `-s /bin/sh` like you suggested.

Any thoughts, @hut, @toonn?

**Comment by toonn** [comment]
Yes to '-s /bin/sh', since I think that'll make it indistinguishable from the non sudo form.

`-E` and `-m` I'm not so sure about. First of all it's backwards incompatible so we need good reason. For example, if you have a mapping to start ranger as root in some directory and you use `:shell`, suddenly instead of your `SHELL` or at least root's `SHELL` you get `sh`?
I'm also not sure whether `-m` is necessary? The manpage specifically mentions `HOME, SHELL, USER and LOGNAME`, maybe it preserves the rest? Something people might actually expect to be preserved is `EDITOR/VISUAL`, you could say just to set it up for root too. But having sudo privileges isn't exactly equal to being root, maybe the admin is crazy and likes emacs, I'd still want vim opened (>: >). Not sure that scenario actually makes sense since we use `su` too though...

**Comment by toonn** [comment]
Maybe we can clear the env with env except for a few special cases like `EDITOR` and keep the flags? Or somehow resolve variables in the command before passing it to sudo, I expect that to be fraught with issues though.
